File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/OpenApiDecoratorTest.java
package io.micronaut.configuration.openapi.docs;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class OpenApiDecoratorTest {

    @Test
    void operationIdsArePrefixedAndSuffixed() {
        var paths = OpenApiSpec.load().getPaths();

        assertEquals("cats-save-suffix", paths.get("/cats").getPost().getOperationId());
        assertEquals("cats-get-suffix", paths.get("/cats/{id}").getGet().getOperationId());
        assertEquals("dogs-save", paths.get("/dogs").getPost().getOperationId());
        assertEquals("dogs-get", paths.get("/dogs/{id}").getGet().getOperationId());
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/openapi/docs/OpenApiDecoratorTest.kt
package io.micronaut.configuration.openapi.docs

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class OpenApiDecoratorTest {

    @Test
    fun operationIdsArePrefixedAndSuffixed() {
        val paths = OpenApiSpec.load().paths

        assertEquals("cats-save-suffix", paths["/cats"]!!.post.operationId)
        assertEquals("cats-get-suffix", paths["/cats/{id}"]!!.get.operationId)
        assertEquals("dogs-save", paths["/dogs"]!!.post.operationId)
        assertEquals("dogs-get", paths["/dogs/{id}"]!!.get.operationId)
    }
}
